src/main.rs

```diff
@@ -1143,12 +1228,41 @@ fn hid_recorder() -> Result<()> {
         None => find_device()?,
     };
 
+    let sysfs = find_sysfs_path(&path)?.canonicalize()?;
+    let hid_id = u32::from_str_radix(
+        sysfs
+            .extension()
+            .unwrap()
+            .to_str()
+            .expect("not a hex value"),
+        16,
+    )
+    .unwrap();
+
+    let skel_builder = HidrecordSkelBuilder::default();
+    let mut open_skel = skel_builder.open().unwrap();
+    let mut open_maps = open_skel.maps_mut();
+
+    let hid_bpf_struct_ops = open_maps.hid_record();
+    let bytes_hid_id: [u8; 4] = hid_id.to_le_bytes();
+
+    if let Some(data) = hid_bpf_struct_ops.initial_value_mut() {
+        data[0..4].copy_from_slice(&bytes_hid_id);
+    }
+
+    let skel = open_skel.load().unwrap();
+
+    let maps = skel.maps();
+
+    // We need to keep _link around or the program gets immediately removed
+    let _link = maps.hid_record().attach_struct_ops()?;
```

Comment:
even if it's not behind a feature I reckon we should still have it behind a commandline argument to avoid confusing users with output that's near-always identical?

Plus right now this would fail if hid-bpf isn't enabled, right?

Possibly other idea: warn or only start printing bpf event data if the data differs from the kernel data?

Reply:
After our meeting this morning: we should only enable the BPF output if there is a bpf loaded attached to the device. This way things will stand out much more when we see the `B:` line.

But adding a command line argument is also interesting because we will be able to request it when we suspect the kernel driver is acting behnd our back.

Reply:
done in the latest push